### ISSUE TITLE
cache: add nickname and object_type_id object cache

### ISSUE THREAD

**Issue by stefanorosanelli** [bedita/bedita#791]
Add cache (using object cache or similar method) for two intensively used methods in frontends:
-  `BEObject::getType / findObjectTypeId` - performs a query like

``` sql
SELECT `BEObject`.`object_type_id` FROM `objects` AS `BEObject` WHERE `BEObject`.`id` = ### LIMIT 1
```
- `BEObject::getIdFromNickname` - performs a query like

``` sql
SELECT `BEObject`.`id` FROM `objects` AS `BEObject` WHERE `nickname` = '###' AND status = ('on') LIMIT 1
```

**Comment by fquffio** [comment]
Such queries are often performed more than once during the same request, so I think that even a runtime cache (i.e. not persisted to filesystem nor in any key-value storage like Redis, just stored in an PHP array [shared across class instances](http://php.net/manual/en/language.variables.scope.php#language.variables.scope.static)) could be a good choice, considering those are very fast queries on MySQL (on my local machine, they usually take 1~2 ms), thus a lookup in a cache engine wouldn't be orders of magnitude faster, I suspect.

**Comment by stefanorosanelli** [comment]
A runtime cache is surely a good idea

Even if fast on MySQL we are calling these queries often _many_ times (hundreds) 
Using a cache could be a significant boost in some scenarios i suspect

Anyway, we could perform some tests/benchmarks to verify - then proceed

**Comment by batopa** [comment]
Merged with pull PR #811
